src/devtools/client/debugger/src/actions/ui.ts

```diff
@@ -99,10 +100,11 @@ export function flashLineRange(location: HighlightedRange): UIThunkAction {
 }
 
 export function copyToClipboard(source: SourceDetails): UIThunkAction {
-  return (dispatch, getState) => {
-    const content = getSourceContent(getState(), source.id);
-    if (content?.value?.type === "text") {
-      copyTextToClipboard(content.value.value);
+  return async (dispatch, getState, { replayClient }) => {
+    const sourceContents = await getSourceContentsAsync(replayClient, source.id);
+
+    if (typeof sourceContents?.contents === "string") {
+      copyTextToClipboard(sourceContents.contents);
```

Comment:
This change seems iffy. Could take _a while_ for source contents to load (if they haven't already loaded) in which case the UX would be kind of weird:
1. Click the copy menu item
2. Go paste the text, seeing either _nothing_ or a stale value
3. Confused, maybe do something else
4. Your clipboard is async updated unexpectedly (possibly wiping out a different thing you'd copied)

I think we should avoid async copy interactions, especially because of 4.

Reply:
Hmm. This only ends up getting called from `useTabContextMenu`.  I think it's fairly reasonable to assume that by the time you've clicked on the tab, the source has been loaded.

If we're _really_ concerned about it, I guess we could do a sync cache check here instead.

Reply:
The Source viewer is optimized for streaming (because of use cases like Andarist's). Source tabs show as soon as we know file names– after which point, you could use the context menu.

Source content streams in (and is parsed) in chunks. This approach to the copy action would wait on the streaming/chunks to finish, which could be a bit of time.